griptape/drivers/memory/conversation/griptape_cloud_conversation_memory_driver.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import json
+import os
+from typing import Optional
+
+from attrs import Factory, define, field
+
+from griptape.artifacts import BaseArtifact
+from griptape.clients import GriptapeCloudApiClient
+from griptape.drivers import BaseConversationMemoryDriver
+from griptape.memory.structure import BaseConversationMemory, ConversationMemory, Run
+from griptape.mixins import ExponentialBackoffMixin
+
+
+@define(kw_only=True)
+class GriptapeCloudConversationMemoryDriver(BaseConversationMemoryDriver, ExponentialBackoffMixin):
+    id: str = field(
+        default=Factory(lambda self: os.getenv("GT_CLOUD_THREAD_ID", self._get_id()), takes_self=True),
+        metadata={"serializable": True},
+    )
+    api_key: str = field(default=None, metadata={"serializable": False})
+    base_url: str = field(default=None, metadata={"serializable": True})
+    structure_run_id: str = field(default=None, metadata={"serializable": True})
+    client: GriptapeCloudApiClient = field(
+        default=Factory(
+            lambda self: GriptapeCloudApiClient(
+                api_key=self.api_key, base_url=self.base_url, structure_run_id=self.structure_run_id
+            ),
+            takes_self=True,
+        ),
+        metadata={"serializable": False},
+    )
+
+    def _get_id(self) -> Optional[str]:
+        thread_id = None
+        for attempt in self.retrying():
```

Comment:
None of the other drivers implement retrying, this should probably be done at the base level in a separate PR.